[CI]: cibuildwheel fails with HTTP 429 when downloading virtualenv.pyz on macOS arm6

### Describe the bug, including details regarding any error messages, version, and platform.

ref1:  https://github.com/apache/incubator-graphar/actions/runs/22903381620/job/66455540109
ref2:  https://github.com/apache/incubator-graphar/pull/904/changes/3b66a94f34e76f15b4b8486a2c8d7014670a0d88

The host Python 3.9 (EOL) used by cibuildwheel lacks built-in retry logic in urllib. When GitHub rate-limits the download of virtualenv.pyz, the request fails immediately with HTTP Error 429: Too Many Requests instead of retrying.

```
      return self._call_chain(*args)
    File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 494, in _call_chain
      result = func(*args)
    File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 641, in http_error_default
      raise HTTPError(req.full_url, code, msg, hdrs, fp)
  urllib.error.HTTPError: HTTP Error 429: Too Many Requests
  Error: Process completed with exit code 1.
```

### Component(s)

Continuous Integration

--- a/.github/workflows/python-wheel-workflow.yml
+++ b/.github/workflows/python-wheel-workflow.yml
@@ -114,6 +114,8 @@ jobs:
       CIBW_PLATFORM: ${{ matrix.os }}
       CIBW_BUILD: "cp39-* cp310-* cp311-* cp312-* cp313-*"
       CIBW_SKIP: "*-musllinux_*"
+      # Use uv on macOS to avoid transient GitHub rate limits when cibuildwheel
+      CIBW_BUILD_FRONTEND: ${{ matrix.os == 'macos' && 'build[uv]' || 'build' }}
       # Pin arch to the matrix platform
       CIBW_ARCHS: ${{ matrix.platform }}
       CIBW_MANYLINUX_X86_64_IMAGE: ${{ matrix.os == 'linux' && format('manylinux{0}', matrix.manylinux) || '' }}
@@ -232,7 +234,11 @@ jobs:
         run: |
           set -euxo pipefail
           python -m pip install --upgrade pip
-          python -m pip install packaging "cibuildwheel>=2.22"
+          if [ "${{ matrix.os }}" = "macos" ]; then
+            python -m pip install packaging "cibuildwheel[uv]>=2.22"
+          else
+            python -m pip install packaging "cibuildwheel>=2.22"
+          fi
           mkdir -p python/dist
           python -m cibuildwheel --output-dir python/dist "$PKGDIR"
 

--- a/.github/workflows/python-wheel-workflow.yml
+++ b/.github/workflows/python-wheel-workflow.yml
@@ -232,7 +232,7 @@ jobs:
         run: |
           set -euxo pipefail
           python -m pip install --upgrade pip
-          python -m pip install packaging cibuildwheel
+          python -m pip install packaging "cibuildwheel>=2.22"
           mkdir -p python/dist
           python -m cibuildwheel --output-dir python/dist "$PKGDIR"
 

--- a/python/README.md
+++ b/python/README.md
@@ -1,12 +1,12 @@
 # GraphAr Python SDK
 
-GraphAr Python SDK provides Python bindings for the GraphAr C++ library, allowing user to work with GraphAr formatted graph data in Python environments. It includes both a high-level API for data manipulation and a command-line interface for common operations.
+GraphAr Python SDK provides Python bindings for the GraphAr C++ library, allowing users to work with GraphAr formatted graph data in Python environments. It includes both a high-level API for data manipulation and a command-line interface for common operations.
 
 ## Installation
 
 ### Prerequisites
 
-- Python >= 3.7
+- Python >= 3.9
 - pip (latest version recommended)
 - CMake >= 3.15 (for building from source)
 - Apache Arrow >= 12.0 (for building from source)

